autogpt_platform/backend/backend/server/v2/store/db.py

```diff
@@ -765,28 +764,20 @@ async def get_agent(
         )
 
         if not store_listing_version or not store_listing_version.Agent:
-            raise fastapi.HTTPException(
-                status_code=404,
-                detail=f"Store listing version {store_listing_version_id} not found",
-            )
+            logger.error(f"Store listing version {store_listing_version_id} not found")
+            return None
 
         graph_id = store_listing_version.agentId
         graph_version = store_listing_version.agentVersion
         graph = await backend.data.graph.get_graph(graph_id, graph_version)
```

Comment:
If you change the include statement on line 762 to `include={"Agent": AGENT_GRAPH_INCLUDE}` then you can omit this second query and the check below.